multi_lidar_calibrator/multi_lidar_calibrator.py

```diff
@@ -90,54 +92,50 @@ def __init__(self):
             )
         self.tf_subscriber = self.create_subscription(
             TFMessage, self.tf_topic, self.tf_callback, 10
         )
-        self.declared_lidars_flag = False
+        self.start_left_lidar_calibration_service = self.create_service(Trigger, "start_left_lidar_calibration", self.start_lidar_calibration_callback)
 
-        # read all the data from files (without ROS)
-        if self.read_pcds_from_file and self.read_tf_from_table:
-            # get LiDAR names as subdirectory names
-            lidar_list = [
-                os.path.splitext(os.path.basename(path))[0]
-                for path in glob.glob(self.pcd_in_dir + "*")
-            ]
-            pcd_paths = dict(zip(lidar_list, [None] * len(lidar_list)))
-            for lidar in lidar_list:
-                self.declare_parameter(lidar, [0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
-                pcd_paths[lidar] = glob.glob(self.pcd_in_dir + lidar + "*")
-            self.lidar_dict = dict(
-                zip(
-                    lidar_list,
-                    [
-                        Lidar(
-                            lidar,
-                            Translation(*self.get_parameter(lidar).value[0:3]),
-                            Rotation(*self.get_parameter(lidar).value[3:], True),
-                        )
-                        for lidar in lidar_list
-                    ],
-                )
-            )
-            for lidar in lidar_list:
-                pcd = o3d.io.read_point_cloud(pcd_paths[lidar][0])
-                for i in range(1, self.frame_count):
-                    pcd += o3d.io.read_point_cloud(pcd_paths[lidar][i])
-                self.lidar_dict[lidar].load_pcd(pcd)
-            self.process_data()
-            exit(0)
-        elif self.read_pcds_from_file and not self.read_tf_from_table:
-            self.get_logger().error("read_tf_from_table is set to False")
-            exit(1)
-        else:
-            self.get_logger().info("waiting for point cloud messages...")
+    def start_lidar_calibration_callback(self, request, response):
+        """Service callback to start the calibration process for the lidar. Defaults to left lidar
+
+        Args:
+            request (_type_): Trigger request
+            response (_type_): Trigger response
+
+        Returns:
+            _type_: Trigger response, success is True if calibration is completed, False otherwise
+        """
+        calibrating_lidars = [lidar for lidar in self.lidar_data.keys()]
+        self.get_logger().info(f"Received request to calibrate {calibrating_lidars}...")
+        
+        if self.is_calibrating:
+            self.get_logger().warn(f"Calibration of {calibrating_lidars} already in progress!")
+            # not(True) = False
+            response.success = not self.is_calibrating 
+            response.message = f"Calibration of {calibrating_lidars} already in progress!"
+            return response
```

Comment:
I don't think we need this guard because we haven't sent the response from the previous one